client/src/unifyfs.c

```diff
@@ -698,8 +687,63 @@ off_t unifyfs_fid_logical_size(int fid)
     if (unifyfs_fid_is_laminated(fid)) {
         return unifyfs_fid_global_size(fid);
     } else {
-        return unifyfs_fid_local_size(fid);
+        /* invoke an rpc to ask the server what the file size is */
+
+        /* get gfid for this file */
+        int gfid = unifyfs_gfid_from_fid(fid);
+
+        /* get file size for this file */
+        size_t filesize;
+        int ret = invoke_client_filesize_rpc(gfid, &filesize);
+        if (ret != UNIFYFS_SUCCESS) {
+            /* failed to get file size */
+            return (off_t)-1;
+        }
+        return (off_t)filesize;
+    }
+}
+
+/* scan list of files and return fid corresponding to target gfid,
+ * returns -1 if not found */
+int TMP_unifyfs_fid_from_gfid(int gfid)
```

Comment:
@MichaelBrim , this is the ugly hack that I added which I meant to ask about.

Reply:
This the one you need? https://github.com/LLNL/UnifyFS/blob/872cb269f23d8e6473ba43032b47e34d5259769b/client/src/unifyfs.c#L553

It looks awful similar to your "hack" though...

Reply:
Yes, thanks.